specfile/changelog.py

```diff
@@ -177,25 +198,42 @@ def assemble(
         evr: Optional[str] = None,
         day_of_month_padding: str = "0",
         append_newline: bool = True,
+        style: ChangelogStyle = ChangelogStyle.standard,
     ) -> "ChangelogEntry":
         """
         Assembles a changelog entry.
 
         Args:
             timestamp: Timestamp of the entry.
                 Supply `datetime` rather than `date` for extended format.
+                The hour is set to 12 for openSUSE style changelog entries, if a
+                date is passed instead of a datetime
             author: Author of the entry.
             content: List of lines forming the content of the entry.
             evr: EVR (epoch, version, release) of the entry.
             day_of_month_padding: Padding to apply to day of month in the timestamp.
             append_newline: Whether the entry should be followed by an empty line.
+            style: Which style of changelog should be created
 
         Returns:
             New instance of `ChangelogEntry` class.
         """
+        if style == ChangelogStyle.openSUSE and not isinstance(
+            timestamp, datetime.datetime
+        ):
+            timestamp = datetime.datetime(
+                year=timestamp.year, month=timestamp.month, day=timestamp.day, hour=12
+            )
+
```

Comment:
I would move this down a bit, where `timestamp` is handled, it doesn't make much sense to me to have it here.